### PR TITLE
SM DDP release notes/changelog files

### PR DESCRIPTION
**Pull request by piyushghai**
*Issue #, if available:*

*Description of changes:*
Added a release notes folder for SM DDP. 

@aaronmarkham @sandeep-krishnamurthy

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/api/training/smd_data_parallel.rst [hunk 1]
```diff
@@ -70,3 +70,12 @@ model.
...
    smd_data_parallel_tensorflow
+
+Latest Updates
+==============
+
+New features, bug fixes, and improvements are regularly made to the SageMaker distributed data parallel library.
+
+To see the the latest changes made to the library, refer to the library
+`Release Notes
+<https://github.com/aws/sagemaker-python-sdk/blob/master/doc/api/training/smd_data_parallel_release_notes/>`_.
```
Comment (metrizable) [inline doc/api/training/smd_data_parallel.rst:81]: PR build failing doc check:

```
/codebuild/output/src818993926/src/github.com/aws/sagemaker-python-sdk/doc/api/training/smd_data_parallel.rst:81: D005 No newline at end of file
```

looks like your [editor/ide](https://thoughtbot.com/blog/no-newline-at-end-of-file) is not properly configured to produce posix-compliant files ([every line must end in a newline character](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206)).

**Review by metrizable — CHANGES_REQUESTED** [review]

**Review by TEChopra1000 — DISMISSED** [review]

File: doc/api/training/smd_data_parallel.rst (hunk 1 above)
Comment (piyushghai) [reply doc/api/training/smd_data_parallel.rst]: Done.

**Comment by piyushghai** [comment]
@metrizable Can you take a look at this PR when you get a chance ? The CI is passing now. Thanks.

File: doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Sagemaker Distributed Data Parallel - Release Notes
+
+- First Release
+- Getting Started
+
+## First Release
+SageMaker's distributed data parallel library extends SageMaker’s training
+capabilities on deep learning models with near-linear scaling efficiency,
+achieving fast time-to-train with minimal code changes.
+SageMaker Distributed Data Parallel :
```
Comment (metrizable) [inline doc/api/training/smd_data_parallel_release_notes/smd_data_parallel_change_log.md:10]:
```suggestion
SageMaker Distributed Data Parallel:
```

**Review by metrizable — APPROVED** [review]

**Pull request merged**